File: backend/Models/Origami.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace origami_backend.Models
{
    public enum Difficulty { Easy, Medium, Hard }
    public class Origami : BaseEntity
    {
        public Guid? UserId { get; set; }
        public string Name { get; set; }
        public string PicturePath { get; set; }
        public string Description { get; set; }
        public string VideoPath { get; set; }
        public Difficulty Difficulty { get; set; }
        public int EstimatedTime { get; set; }

        // relations
        public virtual User User { get; set; }
        public virtual ICollection<Study> Studies { get; set; }
        public virtual ICollection<Step> Steps { get; set; }
        public virtual ICollection<OrigamiComment> OrigamiComments { get; set; }
    }
}

File: frontend/src/app/interfaces/origami.ts
import { Step } from "./step";
import { Comment } from "./comment";

export interface Origami {
    username: string;
    id: string;
    name: string;
    pricturePath: string;
    description: string;
    videoPath: string;
    difficulty: number;
    estimatedTime: number;
    steps: Step[];
    comments: Comment[];
}
